lib/logreader.c

```diff
@@ -573,7 +573,11 @@ log_reader_io_handle_in(gpointer s)
   log_reader_disable_watches(self);
   if ((self->options->flags & LR_THREADED))
     {
-      main_loop_io_worker_job_submit(&self->io_job, NULL);
+      if (!main_loop_io_worker_job_submit(&self->io_job, NULL))
```

Comment:
This can only fail if `main_loop_workers_quit`, but in that case, we should not retry.

If my assumption is correct, this patch depends on work on a different branch.

Reply:
That is possible. Talked IRL with Bazsi, and concluded that this PR is meaningful on its own, but I might have misunderstood something. Sorry if so.

Reply:
We talked about this IRL, registering a task in the debugger would hopefully resolve the issue in a cleaner way.